File: Java/StackWithArray/Stack.java
public class Stack {
    int top;
    int size;
    Node arr[] = null;

    public Stack(int size) {
        this.size = size;
        top = -1;
        arr = new Node[size];
    }

    boolean isEmpty() {
        return top == -1;
    }

    boolean isFull() {
        return top == size;
    }

    Node peek() {
        if (!isEmpty()) {
            return arr[top];
        }
        System.out.println("Stack is empty !");
        return null;
    }

    void push(Node node) {
        if (!isFull()) {
            arr[++top] = node;
        }
        else {
            System.out.println("Stack is Full");
        }
        System.out.println("Node added");
    }


    Node pop() {
        if (!isEmpty()) {
            System.out.println("Node deleted");
            return arr[top--];
        }
        System.out.println("Stack is empty !");
        return null;
    }

    void printStack() {
        if (!isEmpty()) {
            for (int i = top; i > -1; i--) {
                System.out.print(arr[i].val + " ");
            }
            System.out.println();
        }
        else {
            System.out.println("There is nothing to show");
        }
    }
}

File: Python/StackWithArray/stack.py
class Stack:


    def __init__(self):
        self.stack = list()


    def isEmpty(self):
        return len(self.stack) == 0


    def peek(self):
        return self.stack[-1]


    def push(self, val):
        self.stack.append(val)
        print('Node added')


    def pop(self):
        if self.isEmpty():
            print('Stack is empty !')
            return None
        else:
            return self.stack.pop()
        print('Node deleted')


    def printStack(self):
        if self.isEmpty():
            print('There is nothing to print')

        else:
            for i in self.stack:
                print(i, end=' ')
            print()